### ISSUE TITLE
dotcaller broken with latest sklearn

### ISSUE THREAD

**Issue by Phlya** [open2c/cooltools#374]
This is what I get with sklearn 1.1 or 1.1.1:

```
Traceback (most recent call last):
  File "/Users/ilya/opt/anaconda3/envs/wrappers_test/bin/cooltools", line 11, in <module>
    sys.exit(cli())
  File "/Users/ilya/opt/anaconda3/envs/wrappers_test/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Users/ilya/opt/anaconda3/envs/wrappers_test/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/Users/ilya/opt/anaconda3/envs/wrappers_test/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/ilya/opt/anaconda3/envs/wrappers_test/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/ilya/opt/anaconda3/envs/wrappers_test/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/ilya/opt/anaconda3/envs/wrappers_test/lib/python3.9/site-packages/cooltools/cli/dots.py", line 321, in dots
    centroids = api.dotfinder.clustering_step(
  File "/Users/ilya/opt/anaconda3/envs/wrappers_test/lib/python3.9/site-packages/cooltools/api/dotfinder.py", line 1234, in clustering_step
    pixel_clust = clust_2D_pixels(
  File "/Users/ilya/opt/anaconda3/envs/wrappers_test/lib/python3.9/site-packages/cooltools/api/dotfinder.py", line 204, in clust_2D_pixels
    brc.fit(pixels)
  File "/Users/ilya/opt/anaconda3/envs/wrappers_test/lib/python3.9/site-packages/sklearn/cluster/_birch.py", line 548, in fit
    return self._fit(X, partial=False)
  File "/Users/ilya/opt/anaconda3/envs/wrappers_test/lib/python3.9/site-packages/sklearn/cluster/_birch.py", line 611, in _fit
    self._global_clustering(X)
  File "/Users/ilya/opt/anaconda3/envs/wrappers_test/lib/python3.9/site-packages/sklearn/cluster/_birch.py", line 760, in _global_clustering
    self.labels_ = self._predict(X)
  File "/Users/ilya/opt/anaconda3/envs/wrappers_test/lib/python3.9/site-packages/sklearn/cluster/_birch.py", line 694, in _predict
    argmin = pairwise_distances_argmin(
  File "/Users/ilya/opt/anaconda3/envs/wrappers_test/lib/python3.9/site-packages/sklearn/metrics/pairwise.py", line 791, in pairwise_distances_argmin
    indices = PairwiseDistancesArgKmin.compute(
  File "sklearn/metrics/_pairwise_distances_reduction.pyx", line 672, in sklearn.metrics._pairwise_distances_reduction.PairwiseDistancesArgKmin.compute
  File "sklearn/metrics/_pairwise_distances_reduction.pyx", line 1055, in sklearn.metrics._pairwise_distances_reduction.FastEuclideanPairwiseDistancesArgKmin.__init__
  File "sklearn/metrics/_dist_metrics.pyx", line 1300, in sklearn.metrics._dist_metrics.DatasetsPair.get_for
  File "sklearn/metrics/_dist_metrics.pyx", line 1348, in sklearn.metrics._dist_metrics.DenseDenseDatasetsPair.__init__
  File "stringsource", line 658, in View.MemoryView.memoryview_cwrapper
  File "stringsource", line 349, in View.MemoryView.memoryview.__cinit__
ValueError: ndarray is not C-contiguous
```

No issues with sklearn 1.0!

**Comment by sergpolly** [comment]
saw this as well, when tried to run it on the m2 macbook ... we'll see

**Comment by agalitsyna** [comment]
Got this error with sklearn v 1.1.1. Downgrading to 1.0.2 helped. Will be good to fix in general!

**Comment by nvictus** [comment]
This is a sklearn [bug](https://github.com/scikit-learn/scikit-learn/issues/23988) that is being addressed

https://github.com/scikit-learn/scikit-learn/issues/22587

**Comment by jjerphan** [comment]
FYI: https://github.com/scikit-learn/scikit-learn/pull/23990 included in `scikit-learn>=1.1.2` should fix this issue.
